### PR TITLE
Thunkify reusable-blocks store

### PR DESCRIPTION
**Pull request by adamziel**
This PR migrates the reusable-blocks data store to thunks.

I reworked the unit tests to check for outcomes instead of the specific flow we tested for inside of the generator.

**How to test:**
Open the post editor, create a reusable block, reload the page, convert back to static, create another one, delete it, and confirm everything worked as expected.

### CODE REVIEW

File: packages/reusable-blocks/src/store/test/actions.js [hunk 1]
```diff
@@ -1,17 +1,271 @@
...
+			kind: 'postType',
+			name: 'wp_block',
+			label: 'Reusable blocks',
+		},
+	] );
+
+	return registry;
+}
 
 describe( 'Actions', () => {
```
Comment (talldan) [inline packages/reusable-blocks/src/store/test/actions.js:48]: So these seem more like integration tests now that the entire store is being tested. I quite like it, it feels more useful to test stores as a standalone module.

Is it worth still keeping separate unit tests for selectors and reducers given they are also tested here?

File: packages/reusable-blocks/src/store/test/actions.js [hunk 2]
```diff
@@ -1,17 +1,271 @@
...
+			const updatedBlocks = registry
+				.select( blockEditorStore )
+				.getBlocks();
+
+			const newClientId = updatedBlocks[ 0 ].clientId;
+
+			expect( updatedBlocks ).toHaveLength( 1 );
+
+			// Delete random clientId before matching with snapshot
+			delete updatedBlocks[ 0 ].clientId;
```
Comment (talldan) [inline packages/reusable-blocks/src/store/test/actions.js:137]: I had a similar issue in another test and decided to do this:
```js
updatedBlocks[ 0 ].clientId = `[${ typeof updatedBlocks[ 0 ].clientId }]`
```

At least the test can assert the type of the property is correct. Not something required, but just thought I'd share.

**Review by talldan — APPROVED** [review]
Looks great, nice job!

**Review by aristath — APPROVED** [review]
Works as advertised 👍

File: packages/reusable-blocks/src/store/test/actions.js (hunk 1 above)
Comment (adamziel) [reply packages/reusable-blocks/src/store/test/actions.js:48]: > Is it worth still keeping separate unit tests for selectors and reducers given they are also tested here?

I think it is worth it, it creates an easy path of zeroing-in on a problem: imagine the actions-related tests fail, but the selectors-related tests don't. Now we know the selectors are fine.

File: packages/reusable-blocks/src/store/test/actions.js (hunk 2 above)
Comment (adamziel) [reply packages/reusable-blocks/src/store/test/actions.js:137]: I like that idea! I ended up merging this PR as is, but I may follow up with that change. For sure it's something I'll use in the future, thank you!

**Pull request merged**